trace/id_test.go

```diff
@@ -0,0 +1,156 @@
+package trace
+
+import (
+	"log"
+	"testing"
+)
+
+const (
+	TestIDStr = "094f13e642b9b07f"
+	TestIDVal = 670776749184364671
+)
+
+func TestNewID(t *testing.T) {
+	i1 := NewID()
+	i2 := NewID()
+	if i1 == i2 {
+		t.Errorf("Two generated IDs are the same: %d, %d", i1, i2)
+	}
+	if i1 == 0 || i2 == 0 {
+		t.Errorf("One of the ids are 0: %d, %d", i1, i2)
+	}
+}
+
+func TestParseID(t *testing.T) {
+	i, err := ParseID(TestIDStr)
+	if err != nil {
+		t.Error("Error occured parsing the ID: ", err.Error())
+	}
+	if i != TestIDVal {
+		t.Errorf("The parsed value is not matching: %d", i)
+	}
+}
+
+func TestWrongParseID(t *testing.T) {
+	_, err := ParseID("x")
+	if err == nil {
+		t.Error("Error should have happened for wrong input parsing")
+	}
+}
+
+func TestString(t *testing.T) {
```

Comment:
This test method does not say anything about what it tests.

What I usually do is include the type in the test function name so it is obvious what is the subject of the test.